src/evaluators/JSXElement.js

```diff
@@ -0,0 +1,370 @@
+    let trimmedLine = line.replace(/\t/g, " ");
+
+    // trim whitespace touching a newline
+    if (!isFirstLine) {
+      trimmedLine = trimmedLine.replace(/^[ ]+/, "");
+    }
+
+    // trim whitespace touching an endline
+    if (!isLastLine) {
+      trimmedLine = trimmedLine.replace(/[ ]+$/, "");
+    }
+
+    if (trimmedLine) {
+      if (!isLastNonEmptyLine) {
+        trimmedLine += " ";
+      }
+
+      str += trimmedLine;
+    }
+  }
+
+  if (str) {
+    return str;
+  }
+  return null;
+}
+
+function evaluateJSXMemberExpression(ast: BabelNode, strictCode: boolean, env: LexicalEnvironment, realm: Realm) {
+  switch (ast.type) {
+    case "JSXIdentifier":
+      return GetValue(realm, ResolveBinding(realm, ((ast: any): BabelNodeJSXIdentifier).name, strictCode, env));
+    case "JSXMemberExpression":
+      return GetValue(
+        realm,
+        env.evaluate(convertJSXExpressionToIdentifier(((ast: any): BabelNodeJSXMemberExpression)), strictCode)
+      );
+    default:
+      invariant(false, "Unknown JSX Identifier");
+  }
+}
+
+function evaluateJSXIdentifier(ast, strictCode, env, realm) {
+  if (isTagName(ast)) {
+    // special cased lower-case and custom elements
+    return new StringValue(realm, ((ast: any): BabelNodeJSXIdentifier).name);
+  }
+  return evaluateJSXMemberExpression(ast, strictCode, env, realm);
+}
+
+function evaluateJSXValue(value: BabelNode, strictCode: boolean, env: LexicalEnvironment, realm: Realm) {
+  if (value != null) {
+    switch (value.type) {
+      case "JSXText":
+        return new StringValue(realm, ((value: any): BabelNodeJSXText).value);
+      case "StringLiteral":
+        return new StringValue(realm, ((value: any): BabelNodeStringLiteral).value);
+      case "JSXExpressionContainer":
+        return GetValue(realm, env.evaluate(((value: any): BabelNodeJSXExpressionContainer).expression, strictCode));
+      case "JSXElement":
+        return GetValue(realm, env.evaluate(value, strictCode));
+      default:
+        invariant(false, `Unknown JSX value type: ${value.type}`);
+    }
+  }
+  invariant(false, `Null or undefined value passed when trying to evaluate JSX node value`);
+}
+
+function isTagName(ast: BabelNode) {
+  return ast.type === "JSXIdentifier" && /^[a-z]|\-/.test(((ast: any): BabelNodeJSXIdentifier).name);
+}
+
+function getDefaultProps(
+  elementType: BabelNodeJSXIdentifier | BabelNodeJSXMemberExpression,
+  env,
+  realm: Realm
+): null | ObjectValue {
+  let name;
+  if (elementType.type === "JSXIdentifier") {
+    name = elementType.name;
+  }
+  if (!isTagName(elementType) && typeof name === "string") {
+    try {
+      // find the value of "ComponentXXX.defaultProps"
+      let defaultProps = GetValue(
+        realm,
+        env.evaluate(t.memberExpression(t.identifier(name), t.identifier("defaultProps")), false)
+      );
+
+      if (defaultProps instanceof ObjectValue) {
+        return defaultProps;
+      }
+    } catch (error) {
```

Comment:
What is this `catch` trying to put under the carpet? Either a comment or logging an error seems appropriate. No code coverage from tests.

Reply:
If this is abrupt it should probably just let it pass upwards and be handled as an error executing this. So this `catch` isn't necessary.

Did you run into something that needed this?